discord/client.py

```diff
@@ -197,6 +198,32 @@ class Client:
         To enable these events, this must be set to ``True``. Defaults to ``False``.
 
         .. versionadded:: 2.0
+    bucket_storage_cls: type[:class:`.BucketStorageProtocol`]
+        The class to use for storing rate limit buckets given by Discord.
+
+        .. versionadded:: 2.5
+    concurrency: :class:`int`
+        Number of requests that can occur every `per` seconds.
+        This determines your global rate limit prediction.
+
+        Make sure `concurrency` is properly set to maximize the
+        number of requests you can do every `per` seconds. For the
+        majority of bots, this will be the default of `50`, but if Discord
+        increases your `concurrency` make sure to change this value.
+
+        Defaults to `50`.
+
+        .. versionadded:: 2.5
+    per: :class:`float`
+        Number of seconds to wait until resetting `global_concurrency`.
+        Defaults to `1` second.
+
+        .. versionadded:: 2.5
+    rate_limit_timeout: :class:`float`
+        The maximum amount of seconds the client is allowed to wait on a rate limit
+        before raising an exception. Defaults to `-1`, or infinite.
```

Comment:
It would be better if this parameter could accept `None` instead of accepting negative values. Especially since the handling for negative values only checks for -1 and nothing else.